pkg/service/schema.go

```diff
@@ -0,0 +1,83 @@
+package service
+
+// ParameterSchemas is a set of optional JSONSchemas that describe
+// the expected parameters for creation and update of instances and
+// creation of bindings.
+type ParameterSchemas struct {
+	ServiceInstances *InstanceSchema `json:"service_instance,omitempty"`
+	ServiceBindings  *BindingSchema  `json:"service_binding,omitempty"`
+}
+
+// InstanceSchema represents a plan's schemas for creation and
+// update of an API resource.
+type InstanceSchema struct {
+	Create *InputParameters `json:"create,omitempty"`
+	Update *InputParameters `json:"update,omitempty"`
+}
+
+// BindingSchema represents a plan's schemas for the parameters
+// accepted for binding creation.
+type BindingSchema struct {
+	Create *InputParameters `json:"create,omitempty"`
+}
+
+// InputParameters represents a schema for input parameters for creation or
+// update of an API resource.
+type InputParameters struct {
+	Parameters *ParametersSchema `json:"parameters,omitempty"`
+}
+
+// ParametersSchema represents the individual schema for a service
+type ParametersSchema struct {
+	Schema     string                 `json:"$schema"`
+	Type       string                 `json:"type"`
+	Properties map[string]interface{} `json:"properties,omitempty"`
+	Required   []string               `json:"required,omitempty"`
+}
+
+// Parameter represents the individual schema for a given parameter
+type Parameter struct {
```

Comment:
This type encapsulates all the bits of JSON Schema that we need to include AFAICT. The "Items" and "Properties" allow you to embed this object again for arrays. You can find some examples of that in some of the modules (specifically the Firewall Rules where we need to define complex types)